drivers/bluetooth/hci/h4.c

```diff
@@ -35,6 +35,17 @@
 #define H4_EVT  0x04
 #define H4_ISO  0x05
 
+
+/* Extern h4_setup function.
+ * This function executes vendor-specific commands sequence to 
+ * initialize BT Controller before BT Host executes Reset sequence.
+ * h4_setup function must be implemented in vendor-specific HCI 
+ * extansion module if CONFIG_BT_HCI_H4_SETUP is enabled. 
+ */
+#ifdef CONFIG_BT_HCI_H4_SETUP
+extern int h4_setup(void);
```

Comment:
Considering that you needed the `struct device` pointer in your handler, I wonder if we shouldn't add a layer inside `h4.c` which gives this pointer to the vendor-specific callback. In such a design you'd have a `h4_setup()` function always present in `h4.c` and it would then do something like:
```
#if defined(CONFIG_HCI_H4_SETUP)
        return h4_vnd_setup(h4_dev);
#else
        return 0;
#endif
```

Reply:
done.